packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -28,17 +28,31 @@ const getNextCellIndexes = (code: string, indexes: GridCellIndexCoordinates) =>
     throw new Error('Material-UI: The first argument (code) should be an arrow key code.');
   }
 
-  if (code === 'ArrowLeft') {
-    return { ...indexes, colIndex: indexes.colIndex - 1 };
-  }
-  if (code === 'ArrowRight') {
-    return { ...indexes, colIndex: indexes.colIndex + 1 };
-  }
-  if (code === 'ArrowUp') {
-    return { ...indexes, rowIndex: indexes.rowIndex - 1 };
+  switch (code) {
+    case 'ArrowLeft':
+      if (indexes.rowIndex === null) {
+        return { rowIndex: null, colIndex: indexes.colIndex - 1 };
+      }
+      return { ...indexes, colIndex: indexes.colIndex - 1 };
+    case 'ArrowRight':
+      if (indexes.rowIndex === null) {
+        return { rowIndex: null, colIndex: indexes.colIndex + 1 };
+      }
+      return { ...indexes, colIndex: indexes.colIndex + 1 };
+    case 'ArrowUp':
+      if (indexes.rowIndex === 0 || indexes.rowIndex === null) {
+        return { ...indexes, rowIndex: null };
+      }
+
+      return { ...indexes, rowIndex: indexes.rowIndex - 1 };
```

Comment:
return { ...indexes, rowIndex:  indexes.rowIndex ? indexes.rowIndex - 1 : 0};

Reply:
It can't be just 0 because 0 is the first row. So it needs to be able to identify a header somehow.

Reply:
Updated